crates/runtime/src/object_store_registry.rs

```diff
@@ -203,13 +207,201 @@ impl SpiceObjectStoreRegistry {
         )) as Arc<dyn ObjectStore>)
     }
 
+    fn prepare_azure_object_store(url: &Url) -> datafusion::error::Result<Arc<dyn ObjectStore>> {
+        let mut url = url.clone();
+        let params: HashMap<String, String> = parse(url.fragment().unwrap_or_default().as_bytes())
+            .into_owned()
+            .collect();
+        url.set_fragment(None);
+        let mut builder = MicrosoftAzureBuilder::from_env();
+
+        if let Some(sas) = params.get("sas_string") {
+            url.set_query(Some(sas));
+        }
+
+        if let Some(use_emulator) = params.get("use_emulator") {
+            let as_bool = use_emulator.parse::<bool>().map_err(|_| {
+                DataFusionError::Configuration(format!(
+                    "{} is not a valid boolean for use_emulator",
+                    use_emulator
+                ))
+            })?;
+            builder = builder.with_use_emulator(as_bool);
+            if !as_bool {
+                builder = builder.with_url(url.to_string());
+            }
+        } else {
+            builder = builder.with_url(url.to_string());
+        }
+
+        builder = builder.with_url(url.to_string());
+
+        if let Some(account) = params.get("account") {
```

Comment:
Nit: If there are some required combination of parameters, like `skip_signature` if there are no credentials specified, we should help user to understand the problem (use friendly error). Will MicrosoftAzureBuilder handle this nicely?

Reply:
I've added a condition to the actual connector itself to test for this connection, we'll throw an error if there's a incorrect combination of credentials.